### PR TITLE
[Docs] Add faq in dev-1.x branch

### PR DESCRIPTION
**Pull request by Renzhihan**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by MeowZheng** [comment]
The faq is in docs/en/notes/faq.md for mmseg 1.x, and I think it is not necessary to add the version dependency of mmseg 0.x

**Comment by Renzhihan** [comment]
> The faq is in docs/en/notes/faq.md for mmseg 1.x, and I think it is not necessary to add the version dependency of mmseg 0.x

Thanks for your reply, I made a mistake in understanding the task. I will update the docs/en/notes/faq.md later.

**Review by MeowZheng — COMMENTED** [review]
please follow #2768 to add the version dependency of 1.x

**Comment by Renzhihan** [comment]
> 

I follow https://github.com/open-mmlab/mmsegmentation/pull/2769 to delete the "Why is the log file not created" and other version errors.

File: docs/en/notes/faq.md [hunk 1]
```diff
@@ -26,5 +26,86 @@ Notes:
```
Comment (csatsurnh) [inline docs/en/notes/faq.md:26]: please add notes according to zh_cn faq
https://github.com/open-mmlab/mmsegmentation/blob/1896344e04ad3a919194b394d44da4e0b69d026b/docs/zh_cn/notes/faq.md?plain=1#L21-L25

File: docs/en/notes/faq.md [hunk 2]
```diff
@@ -24,7 +24,95 @@ Notes:
...
 - Infer from the log file. Open the log file of the model and search `nGPU` in the file. The number of figures following `nGPU` is the number of GPUs needed to train the model. For instance, searching for `nGPU` in the log file yields the record `nGPU 0,1,2,3,4,5,6,7`, which indicates that eight GPUs are needed to train the model.
+
+## What does the auxiliary head mean
+
+Briefly, it is a deep supervision trick to improve the accuracy. In the training phase, `decode_head` is for decoding semantic segmentation output, `auxiliary_head` is just adding an auxiliary loss, the segmentation result produced by it has no impact to your model's result, it just works in training. You may read this [paper](https://arxiv.org/pdf/1612.01105.pdf) for more information.
+
+## How to output the image for painting the segmentation mask when running the test script
+
+In the test script, we provide `show-dir` argument to control whether output the painted images. Users might run the following command:
+
```
Comment (xiexinch) [inline docs/en/notes/faq.md:46]:
```suggestion
In the test script, we provide `--out` argument to control whether output the painted images. Users might run the following command:

```

File: docs/en/notes/faq.md [hunk 3]
```diff
@@ -24,7 +24,95 @@ Notes:
...
 - Infer from the log file. Open the log file of the model and search `nGPU` in the file. The number of figures following `nGPU` is the number of GPUs needed to train the model. For instance, searching for `nGPU` in the log file yields the record `nGPU 0,1,2,3,4,5,6,7`, which indicates that eight GPUs are needed to train the model.
+
+## What does the auxiliary head mean
+
+Briefly, it is a deep supervision trick to improve the accuracy. In the training phase, `decode_head` is for decoding semantic segmentation output, `auxiliary_head` is just adding an auxiliary loss, the segmentation result produced by it has no impact to your model's result, it just works in training. You may read this [paper](https://arxiv.org/pdf/1612.01105.pdf) for more information.
+
+## How to output the image for painting the segmentation mask when running the test script
+
```
Comment (xiexinch) [inline docs/en/notes/faq.md:44]:
```suggestion
## How to output the segmentation mask image when running the test script

```

File: docs/en/notes/faq.md [hunk 4]
```diff
@@ -24,7 +24,95 @@ Notes:
...
+```python
+if self.reduce_zero_label:
+    # avoid using underflow conversion
+    gt_semantic_seg[gt_semantic_seg == 0] = 255
+    gt_semantic_seg = gt_semantic_seg - 1
+    gt_semantic_seg[gt_semantic_seg == 254] = 255
+```
+
+**Noted:** Please pay attention to label numbers of dataset when using `reduce_zero_label`. If dataset only has two types of labels (i.e., label 0 and 1), it needs to close `reduce_zero_label`, i.e., set `reduce_zero_label=False`.
```
Comment (xiexinch) [inline docs/en/notes/faq.md:118]: Please refer to https://github.com/open-mmlab/mmsegmentation/blob/dev-1.x/docs/zh_cn/notes/faq.md

**Review by xiexinch — APPROVED** [review]

**Pull request merged**